include/deal.II/base/bounding_box.h

```diff
@@ -0,0 +1,108 @@
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_base_bounding_box_h
+#define dealii_base_bounding_box_h
+
+
+#include <deal.II/base/config.h>
+#include <deal.II/base/exceptions.h>
+#include <deal.II/base/point.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A class that represents a bounding box in a space with arbitrary dimension
+ * <tt>spacedim</tt>.
+ *
+ * Objects of this class are used to represent bounding boxes. They are,
+ * among other uses, useful in parallel distributed meshes to give a general
+ * description the owners of each portion of the mesh.
+ *
+ * Boundig boxes are represented by two vertices and, geometrically, are
+ * represented by the following:
+ * 1 d: a segment (represented by its vertices)
+ * 2 d: a rectangle (represented by two vertices V opposed to the center)
+ * V--------
+ * |        |
+ * ---------V
+ * 3 d: a cuboid (represented by two vertices V opposed to the center
+ *               and not owned by the same face)
+ *   V------.
+ *  /      /|
+ * .------. |
+ * |      | /
+ * |      |/
+ * .------V
+ *
+ * Notice the sides are always parallel to the respective axis.
+ * 
+ * @author Giovanni Alzetta, 2017
+ */
+template <int spacedim, typename Number=double>
+class BoundingBox
+{
+public:
+  /**
+   * Standard constructor. Creates an object that corresponds to a null box,
+   * i.e. an degenerate box with both points being the origin.
+   */
+  BoundingBox ();
+
+  /**
+   * Standard constructor for non-empty boxes: it uses a pair of points
+   * which describe the box: one for the bottom and one for the top
+   * corner.
+   */
+  BoundingBox (const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &boundary_points); //Add two points constructor
+
+  /**
+   * Returns the boundary_points
+   */
+  std::pair<Point<spacedim,Number>,Point<spacedim,Number>> & get_boundary_points ();
+
+  /**
+   * Returns true if the point is inside the Bounding Box, false otherwise
+   */
+  bool point_inside (const Point<spacedim, Number> &p) const;
+
+private:
+  std::pair<Point<spacedim, Number>,Point<spacedim,Number>> boundary_points;
+};
+
+/*------------------------------- Inline functions: Point ---------------------------*/
+
+#ifndef DOXYGEN
+
+// At least clang-3.7 requires us to have a user-defined constructor
+// and we can't use 'BoundingBox<int spacedim>::BoundingBox ()' here.
+template <int spacedim, typename Number >
+inline
+BoundingBox<spacedim, Number>::BoundingBox ()
+{
+}
+
+
+template <int spacedim, typename Number>
+inline
+BoundingBox<spacedim, Number>::BoundingBox (const std::pair<Point<spacedim,Number>,Point<spacedim,Number>> &boundary_points)
+{
+    this->boundary_points = boundary_points;
```

Comment:
i think you should check that points are correct.

Reply:
Thank you for the comments: they are very useful!!!
As for this one I wonder: this seems a right place to check if points generate a bounding box of dimension spacedim. Do you think this is the sort of test I should do?

The problem with an assert statement here is that dimension is guaranteed only when you create a bounding box giving a pair of points while with the default constructor you're guaranteed the bounding box is, in reality, a point...would this be ok?

Thank you very much for your help!

Reply:
>  Do you think this is the sort of test I should do?

yes, you need to check it here. Since you don't call this constructor from the default, it should be no problem.